src/metrics.cpp

```diff
@@ -436,20 +444,33 @@ int printMessageBox(size_t cols)
 
 int printInitMessage()
 {
-    if (loaded) {
+    static bool doneLoadingPrinted = false;
+
+    if (loaded && doneLoadingPrinted && !fImporting) {
         return 0;
     }
 
-    std::string msg = *initMessage;
-    std::cout << _("Init message:") << " " << msg << std::endl;
-    std::cout << std::endl;
-
-    if (msg == _("Done loading")) {
-        loaded = true;
+    std::string msg;
+    if (!loaded)
+    {
+        msg = static_cast<std::string>(_("Init message: ")) + static_cast<std::string>(*initMessage);
+    }
+    else if (loaded && !doneLoadingPrinted)
+    {
+        msg = static_cast<std::string>(_("Init message: Done loading"));
+        doneLoadingPrinted = true;
+    }
+    else if (fImporting)
+    {
+        msg = static_cast<std::string>(_("Init message (node already loaded): ")) + static_cast<std::string>(*initMessageAfterLoading);
     }
 
+    std::cout << msg << std::endl;
+    std::cout << std::endl;
```

Comment:
std::endl means "append new line char and flush stream"
This should be written as 
```suggestion
    std::cout << msg << "\n" << std::endl;
```

Reply:
Agree with your suggestion, modified accordingly.